lib/iris/tests/__init__.py

```diff
@@ -695,18 +704,38 @@ def check_graphic(self, tol=_DEFAULT_IMAGE_TOLERANCE):
 
             figure.savefig(result_fname)
 
-            if not os.path.exists(expected_fname):
-                warnings.warn('Created image for test %s' % unique_id)
-                shutil.copy2(result_fname, expected_fname)
+            # XXX: Deal with a new test result i.e. it's not in the repo
 
-            err = mcompare.compare_images(expected_fname, result_fname, tol=tol)
+            # hash the created image using sha1
+            with open(result_fname, 'rb') as res_file:
+                sha1 = hashlib.sha1(res_file.read())
+            if unique_id not in repo:
+                msg = 'Image comparison failed: Created image {} for test {}.'
+                raise ValueError(msg.format(result_fname, unique_id))
```

Comment:
@marqh You've changed the logic here from the original, which I disagree with.

As it now stands any developer who adds a **new** graphics test, has to manually craft the changes to the `imagerepo.json` and manage the hash renaming of the `png` ... this doesn't scale very well.

I'll propose an alternative approach on your branch as a PR that (I believe) improves this work-flow. We can discuss that further there ... so hang tight for that.